### ISSUE TITLE
🐛 BUG:

### ISSUE THREAD

**Issue by djmtype** [withastro/astro#3647]
### What version of `astro` are you using?

1.0.0-beta.46

### Are you using an SSR adapter? If so, which one?

none

### What package manager are you using?

npm

### What operating system are you using?

mac

### Describe the Bug

Vite allows for CSS sourcemaps https://vitejs.dev/config/#css-devsourcemap
Prior to Astro 1.0.0 beta 42, CSS sourcemaps in Astro worked as advertised. 
By configuring Vite within astro.config.mjs this is made possible: 
```js
vite: {
    css: {
      devSourcemap: true,
    },
  },
```

Now the references to the css files are all broken. Upon inspection, the reference to the css property always points to the page name instead of pointing to the .css or .astro component file – where the css styles originate. 

In devtools, inspect any html property with styles applied, and it will show you the page as the reference instead of the css file where the style resides.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-3d931t

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by bholmesdev** [comment]
Thanks for reporting! I took a look at your example, but I'm not seeing any bad network requests or other errors in the console. Are there any logs in particular we should be looking for here?

**Comment by djmtype** [comment]
A few RCs ago, this bug was kind of fixed, meaning CSS sourcemaps are now present. However, the references are also duplicated which can make debugging still difficult. 

For example, on an astro file, I get the same css style listed under both the component source and the page source.
![Screen Shot 2022-08-09 at 1 05 42 PM](https://user-images.githubusercontent.com/930713/183714327-5baad5be-6b3d-419d-b404-13a187535fe9.png)

**Comment by bholmesdev** [comment]
Yep, I've noticed this as well! Definitely confusing to debug. Hoping it's a duplicate sourcemap we're merging somewhere up the chanin.

**Comment by supermoos** [comment]
To me it doesn't even point to file names, but just in-page `<style>` tags.
<img width="455" alt="Screenshot 2022-09-14 at 14 57 50" src="https://user-images.githubusercontent.com/2298345/190160254-6f008481-d7f8-4577-8401-25cf220e4b81.png">

**Comment by supermoos** [comment]
Adding the config from the first post helps, but then duplicates the styles:
<img width="457" alt="Screenshot 2022-09-14 at 15 00 36" src="https://user-images.githubusercontent.com/2298345/190160791-3b38762c-f3f6-4f87-b327-d6c9f5806b13.png">

**Comment by matthewp** [comment]
Just checked and sourcemaps are working now! Let's treat the duplication as a separate issue. Thanks!

**Comment by djmtype** [comment]
@matthewp My initial filing of this bug is actually the **duplication** of sourcemaps. 
The CSS sourcemaps always worked, however they duplicated at the page-level, and should only exist at the file-level.

**Comment by matthewp** [comment]
@djmtype the original issue comment didn't speak to duplication. I see what is happening and I agree it's issue worthy, it would just be good to have a new one as duplication and sourcemaps not pointing to the right file are different issues.

**Comment by djmtype** [comment]
👍 Gotcha @matthewp
